src/runtime_src/core/edge/user/aie_sys.h

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (C) 2021 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef _XCL_AIE_SYS_H_
+#define _XCL_AIE_SYS_H_
+
+#include <string>
+#include <vector>
+#include <fstream>
+
+namespace aie_sys_parser {
+
+static std::fstream sysfs_open_path(const std::string& path, std::string& err,
+    bool write, bool binary)
+{   
+    std::fstream fs;
+    std::ios::openmode mode = write ? std::ios::out : std::ios::in;
+    
+    if (binary) 
+        mode |= std::ios::binary;
+    
+    err.clear();
+    fs.open(path, mode);
+    if (!fs.is_open()) {
+        std::stringstream ss;
+        ss << "Failed to open " << path << " for "
+            << (binary ? "binary " : "")
+            << (write ? "writing" : "reading") << ": "
+            << strerror(errno) << std::endl;
+        err = ss.str();
+    }
+    return fs;
+}
+
+std::fstream sysfs_open(const std::string& entry,
+    std::string& err, bool write, bool binary)
+{
+    return sysfs_open_path(entry, err, write, binary);
+}
+
+void sysfs_get(const std::string& entry, std::string& err_msg,
+    std::vector<std::string>& sv)
+{
+    std::fstream fs = sysfs_open(entry, err_msg, false, false);
+    if (!err_msg.empty())
+        return;
+
+    sv.clear();
+    std::string line;
+    while (std::getline(fs, line))
+        sv.push_back(line);
+}
+
+void addrecursive(int col, int row, std::string tag, std::string line, boost::property_tree::ptree &pt) {
```

Comment:
std::string are passed by value?  I think you want to pass by const std::string& unless you are using the arguments as local r/w variables.

Reply:
done